src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/OrderedApiColumnDefContainer.java

```diff
@@ -0,0 +1,17 @@
+package io.stargate.sgv2.jsonapi.service.schema.tables;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import java.util.LinkedHashMap;
+
+/** A {@link ApiColumnDefContainer} that maintains the order of the columns as they were added. */
+public class OrderedApiColumnDefContainer extends LinkedHashMap<CqlIdentifier, ApiColumnDef>
```

Comment:
As with shredding, I am not sure there is any value in separating ordered and unordered: instead may just implement and use ordered always, reduce number of types slightly.